bothan-api/server/src/monitoring/client.rs

```diff
@@ -0,0 +1,72 @@
+use reqwest::header::HeaderMap;
+use reqwest::Response;
+use serde::Serialize;
+
+use bothan_core::store::ActiveSignalIDs;
+
+use crate::monitoring::error::PostError;
+use crate::monitoring::signer::Signer;
+use crate::monitoring::types::{BothanInfo, Entry, Topic};
+use crate::proto::price::Price;
+use crate::VERSION;
+
+pub struct Client {
+    url: String,
+    client: reqwest::Client,
+    signer: Signer,
+}
+
+impl Client {
+    pub fn new<T: Into<String>>(url: T, signer: Signer) -> Self {
+        Self {
+            url: url.into(),
+            client: reqwest::Client::new(),
+            signer,
+        }
+    }
+
+    pub async fn post_price(
+        &self,
+        uuid: String,
+        prices: Vec<Price>,
+    ) -> Result<Response, PostError> {
+        self.post(uuid, Topic::Price, prices).await
+    }
+
+    pub async fn post_heartbeat(
+        &self,
+        uuid: String,
+        active_signal_ids: ActiveSignalIDs,
+    ) -> Result<Response, PostError> {
+        let bothan_info = BothanInfo::new(
+            VERSION.to_string(),
+            active_signal_ids.into_iter().collect::<Vec<String>>(),
```

Comment:
So should we add supported sources in this too?

Reply:
I think we've decided to keep that in telemetry instead rather than the public monitoring section